File: Lesson.03 - Design Patterns - Structural/Bridge/Gmail.java
public class Gmail implements EmailProvider {
    @Override
    public void authenticate() {
        System.out.println("Gmail: Authenticating with OAuth tokens");
    }

    @Override
    public void processEmail(String emailData) {
        System.out.println("Gmail: Processing email through Google infrastructure");
    }

    @Override
    public void logout() {
        System.out.println("Gmail: Logging out from session");
    }
}

File: Structural - Exercises/Solutions-Leonie/Bridge/providers/gmail.py
from providers.provider_interface import ProviderInterface

class GmailProvider(ProviderInterface):
    def authenticate(self):
        print("Gmail: Authenticating with Google tokens")

    def process_email(self, email_data):
        print(f"Processing email through Google infrastructure")

    def logout(self):
        print("Gmail: Logging out from session")
